cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewStatus.java

```diff
@@ -30,13 +32,15 @@
     RUNNING,
     COMPLETED,
     DEPLOY_FAILED,
-    RUN_FAILED
+    RUN_FAILED,
+    KILLED
   }
 
   private final Status status;
-  private final Throwable throwable;
+  // BasicThrowable is required here so that the exception can be serialized in REST response.
+  private final BasicThrowable throwable;
```

Comment:
Since this class is only used in the REST response and not programmatically, using `BasicThrowable` here. We already have a `BasicThrowableCodec` which is being used in `PreviewHttpHandler`.